### ISSUE TITLE
Missing test for mapped arguments property descriptor

### ISSUE THREAD

**Issue by caiolima** [tc39/test262#814]
Arguments object mapping rules are being well covered on test262, however there is no test case covering arguments property descriptor behavior. They are important, since it is not uncommon implement the mapping rules and forget about descriptor changes. One example of case that it can catch this kind of problem is:

```
function foo(x) {
    Object.defineProperty(arguments, 0, {configurable: true, enumerable: false, writable:true, value:42});
    return [x, arguments[0], arguments]
}

var result = foo(1);
if (Object.getOwnPropertyDescriptor(result[2], 0).enumerable === true)
    throw new Error();
```
In this case, the arguments is still mapped and in this case and the ```enumerable``` property can be ignored. We identified this behavior on WebKit JavaScriptCore implementation and fixed on https://bugs.webkit.org/show_bug.cgi?id=159398.

**Comment by bakkot** [comment]
See also #729.

**Comment by leobalter** [comment]
this is fixed by #815
